src/main/java/org/opendevstack/provision/controller/ProjectApiController.java

```diff
@@ -195,6 +196,9 @@
       mailAdapter.notifyUsersAboutProject(newProject);
 
       return ResponseEntity.ok().body(newProject);
+    } catch (ProjectAlreadyExistsException exAlreadyExists) {
+      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+          .body(exAlreadyExists.getMessage());
     } catch (Exception exProvisionNew) {
       Map<CLEANUP_LEFTOVER_COMPONENTS, Integer> cleanupResults =
```

Comment:
Potentially any other exception than ```ProjectAlreadyExistsException``` happening above will lead to this catch block. While this fix and this one #330 helps to reduce the number of unexpected situations, the ```cleanup()``` could still be unexpected triggered. For example, any change/mistake in future implementation of the used dependencies whithin the try/catch block (e.g. mailAdapter, JiraAdatper, etc...) could thrown an exception and trigger the ```cleanup`` method. This change as a bug fix looks ok for me so far.
Additionally, I think we need redesign a bit the project provision flow to make it more resilient and operations ready. I propose therefore (and I'll create the corresponding issue) to: 
- not remove left over components in case of exception" but mark them for deletion and 
- wrap the project creation flow in a task
- create a screen that shows list of tasks with its status (error description), so admin do not need to dig into the log file to find the error descriptoin
- provide posibility(buttons) for retry (complete task) or cleanup
@clemensutschig @metmajer your thoughs?